src/unix/linux.c

```diff
@@ -1343,47 +1343,163 @@ static uint64_t uv__read_uint64(const char* filename) {
 }
 
 
-/* This might return 0 if there was a problem getting the memory limit from
- * cgroups. This is OK because a return value of 0 signifies that the memory
- * limit is unknown.
- */
-static uint64_t uv__get_constrained_memory_fallback(void) {
-  return uv__read_uint64("/sys/fs/cgroup/memory/memory.limit_in_bytes");
-}
+static void uv__get_cgroup1_memory_limits(char buf[static 1024], uint64_t* high,
+                                          uint64_t* max) {
+  char filename[4097];
+  char* p;
+  int n;
+
+  /* Seek to the memory controller line. */
+  p = strchr(buf, ':');
+  while (p != NULL && strncmp(p, ":memory:", 8)) {
+    p = strchr(p, '\n');
+    if (p != NULL)
+      p = strchr(p, ':');
+  }
 
+  if (p != NULL) {
+    /* Find out where the controller is mounted. */
+    p = p + 9;  /* skip :memory:/ */
```

Comment:
```suggestion
    p = p + sizeof(":memory:");
```
Just a suggestion but it's self-documenting.

Reply:
Ah yeah, that's neat. I used `strlen` instead so that we don't have to do `-1` or remove the trailing `/` (I verified that this is optimized away with my host gcc compiler).